### PR TITLE
Add Sort for FundCampaigns & modify Pledge Update/Delete Resolvers

### LINKED ISSUE #2305: Add Search/Sort & Refac for Fund, Campaigns & Pledge resolvers
**Is your feature request related to a problem? Please describe.**

Query resolvers: `getFundById` `getFundraisingCampaignById` `getFundraisingCampaignPledgeById`

- Add search by & sort functionality to query resolvers for Funds, Fundraising campaigns and associated pledges.
- Bug: Admin can create multiple pledges on behalf of single user for a fundraising campaign (NOTE: Each campaign must only have one pledge per person or family [Reference](https://github.com/PalisadoesFoundation/talawa-api/issues/1663#issue-2071259746)). 
- The current structure of nested resolvers could be improvised by refactoring as they perform unnecessary operations to populate documents from database per request

**Describe the solution you'd like**
- Add sort (ascending/descending) by amount, startDate, endDate, etc for funds, fundraising campaigns and pledges.
- Add search by name, creator (for fund) and pledge volunteer name, etc.

### PR DESCRIPTION
**Pull request by GlenDsza**
**What kind of change does this PR introduce?**

bugfix, feature, refactoring

**Issue Number:**

Fixes #2305 

**Did you add tests for your changes?**

Yes 

**Summary**

- Add sort functionality to `getFundraisingCampaignById` resolver.
- Replace `campaignPledges` and `parentFund` resolvers with populate method.
- Add users[] to `UpdateFundCampaignPledgeInput` to handle update to multiple users participating in pledge
- Remove User check from `removeFundraisingCampaingPledge.ts` & `updateFundCampaignPledge.ts` to enable admin to modify pledges on behalf of user.

**Does this PR introduce a breaking change?**

No

**Have you read the [contributing guide](https://github.com/PalisadoesFoundation/talawa-api/blob/master/CONTRIBUTING.md)?**

Yes 

## Summary by CodeRabbit

- **New Features**
  - Added sorting functionality for pledges in fundraising campaigns.
  - Introduced new sorting options for pledges by amount, start date, and end date.

- **Bug Fixes**
  - Improved date validation logic to handle optional start and end dates.

- **Tests**
  - Updated test cases to reflect new sorting options and removed redundant user authorization checks.
  - Added new test cases for invalid user ID handling and invalid start date validation.

- **Chores**
  - Streamlined test setup by removing unnecessary imports and test cases.

### CODE REVIEW

**Comment by Cioppolo14** [comment]
@GlenDsza Please resolve the coderabbit comments and the failed checks.

File: schema.graphql [hunk 1]
```diff
@@ -1700,6 +1709,7 @@ input UpdateFundCampaignPledgeInput {
   currency: Currency
   endDate: Date
   startDate: Date
+  users: [ID]
```
[the remark this answers is not in the document]
Comment (GlenDsza) [reply schema.graphql:1712]: Updated code & test to check modification of users in `updateFundraisingCampaignPledge`

File: src/resolvers/Mutation/updateFundCampaignPledge.ts [hunk 2]
```diff
@@ -69,17 +68,6 @@ export const updateFundraisingCampaignPledge: MutationResolvers["updateFundraisi
...
-    if (!pledgeUserIds.includes(context.userId)) {
-      throw new errors.ConflictError(
-        requestContext.translate(USER_NOT_MADE_PLEDGE_ERROR.MESSAGE),
-        USER_NOT_MADE_PLEDGE_ERROR.CODE,
-        USER_NOT_MADE_PLEDGE_ERROR.PARAM,
-      );
-    }
 
     let startDate;
     let endDate;
```
[the remark this answers is not in the document]
Comment (GlenDsza) [reply src/resolvers/Mutation/updateFundCampaignPledge.ts]: Done

File: src/resolvers/Query/getFundraisingCampaign.ts [hunk 3]
```diff
@@ -3,6 +3,7 @@ import {
   type InterfaceFundraisingCampaign,
 } from "../../models";
 import type { QueryResolvers } from "../../types/generatedGraphQLTypes";
+import { getSort } from "./helperFunctions/getSort";
```
[the remark this answers is not in the document]
Comment (GlenDsza) [reply src/resolvers/Query/getFundraisingCampaign.ts:6]: `getSort` Error handling probably be handled in separate issue

**Review by palisadoes — APPROVED** [review]

**Pull request merged**